answers/Tezos.md

```diff
@@ -0,0 +1,70 @@
+
+The process of adding new blocks to Tezos is referred to as Baking, and so Bakers are the only actors who participate in creating new blocks and voting on consensus rule changes. There is a minimum "roll size" (number of XTZ held) requirement to be a baker (initially 10k XTZ, will likely be 8k XTZ after the completion of the first upgrade cycle) - and so Tezos' consensus mechanism can be considered to be a form of PoS. Delegation is also baked into the Tezos protocol, and holders who do not wish to bake themselves can delegate their stake to a Baker and receive a share of their rewards. Tezos could therefore be considered to have Delegated Proof of Stake (DPoS), but without the fixed number of block producers and election dynamic usually associated with DPoS (the number of possible Bakers in Tezos is much larger). The Tezos implementation of PoS has been referred to as [Liquid Proof of Stake](https://medium.com/tezos/liquid-proof-of-stake-aec2f7ef1da7).
+
+* *Which entities interact with the blockchain? What role does each entity play? How are the "block producers" selected?*
+
+[Bakers](https://tezos.gitlab.io/mainnet/whitedoc/proof_of_stake.html) are the only entities that participate in creating new blocks, they do so by baking a new block when randomly selected, and by endorsing the new blocks of other bakers (a single block can have many endorsers, and the reward for endorsing is smaller than for baking). Through this method of baking and endorsing, rival blocks engage in short-term competition but are quickly locked in as subsequent blocks are baked. When proposing or endorsing new blocks, a Baker puts up some of their stake as a security deposit - if other bakers observe them double baking or endorsing (and possibly other types of bad behavior) they can be accused and lose this security deposit (with half going to the accuser). 
+
+* *How are changes to the consensus rules decided on and implemented?*
+
+Bakers participate in a four-stage voting [process](https://medium.com/tezos/amending-tezos-b77949d97e1e) to adopt changes to the consensus rules. The amendment process has 4 phases, each lasting 8 "cycles" (a cycle lasts around 3 days, so each phase lasts for around 24 days). In the first phase bakers submit and upvote proposals. When the first phase ends, the most upvoted proposal will progress to phase 2, where it must be approved by at least 80% of bakers. If the criteria are met, this is followed by a testing phase in which a testnet fork with the changes applied is created and runs for 48 hours (a further testnet matching the proposal may be run for the rest of this phase to allow further testing). After the testing phase, bakers vote on whether the changes should be activated, with an 80% supermajority required (adjusts dynamically based on previous participation rates). After this 4th phase the changes are activated (or not) and the cycle begins again with new proposals.
```

Comment:
One remaining question is how, technically, consensus rules are updated. Tezos' language seems to imply it works as it does in PoW, with nodes voluntarily updating, but according to davecgh updates are accomplished via a master key, defeating the whole purpose. Would be nice to know how it actually happens. Perhaps the voting can veto changes made via master key?
https://matrix.to/#/!kdpEDksmOMNrlMqffD:decred.org/$155206988021084WhHYu:decred.org?via=decred.org&via=matrix.org

https://matrix.to/#/!kdpEDksmOMNrlMqffD:decred.org/$155207103621108FyNeQ:decred.org?via=decred.org&via=matrix.org